### ISSUE TITLE
Storybook build failing in plugin PRs

### ISSUE THREAD

**Issue by Rugvip** [spotify/backstage#685]
## Expected Behavior

The storybook builds should not fail when there are no changes to storybook or it's dependencies.

## Current Behavior

Storybook verification build is failing with `Error: Can't resolve 'babel-loader' in '/home/runner/work/backstage/backstage/packages/storybook'`, see e.g. https://github.com/spotify/backstage/pull/651/checks?check_run_id=627548535

## Possible Solution

Might be an issue with the cache, might be an issue with deps sometimes not being built, or something else.

## Steps to Reproduce

? - Maybe this could be reproduced locally by adding a plugin, clean out the repo from dist and node_modules dirs, and then try to run tasks similar to what we're doing with the GitHub actions.

## Context

This also happened for the tech radar PR, where it was possibly fixed by first adding deps and then removing them. https://github.com/spotify/backstage/pull/628

## Your Environment

Build in GitHub Actions

**Comment by OrkoHunter** [comment]
Also for #681, but the error message seems to be different in this https://github.com/spotify/backstage/runs/634375942

```
ERR! Module not found: Error: Can't resolve '@backstage/core' in '/home/runner/work/backstage/backstage/packages/storybook/.storybook'
(node:3228) UnhandledPromiseRejectionWarning: ModuleNotFoundError: Module not found: Error: Can't resolve '@backstage/core' in '/home/runner/work/backstage/backstage/packages/storybook/.storybook'
```

**Comment by Rugvip** [comment]
@OrkoHunter yeah that's to do with @backstage/core not being built in storybook CI builds, but we actually don't want to do that as we'd end up depending on 2 different instances of core. Fixed it in #695 by resolving @backstage/core to it's src dir.

Regarding this issue It seems to have been caused by babel-loader being installed inside react-scripts in node_modules. When clearing yarn.lock it seems like it gets lifted up a level somehow, and that solves the issue. The fix I opted for in #695 was to just get rid of bable-loader altogether, to closer mirror the webpack config used by plugin:serve (and soon app:serve/build).
